src-extras/Database/LSMTree/Extras/Orphans.hs

```diff
@@ -25,13 +39,101 @@ import           System.Random (Uniform)
 deriving anyclass instance Uniform Word256
 
 instance SerialiseKey Word256 where
-  serialiseKey (Word256{word256hi, word256m1, word256m0, word256lo}) =
+  serialiseKey w256 =
     RB.RawBytes $ mkPrimVector 0 32 $ P.runByteArray $ do
       ba <- P.newByteArray 32
-      P.writeByteArray ba 0 $ byteSwap64 word256hi
-      P.writeByteArray ba 1 $ byteSwap64 word256m1
-      P.writeByteArray ba 2 $ byteSwap64 word256m0
-      P.writeByteArray ba 3 $ byteSwap64 word256lo
+      P.writeByteArray ba 0 $ byteSwapWord256 w256
       return ba
+  deserialiseKey (RawBytes (PV.Vector off len ba))
+    | len >= 32  = byteSwapWord256 $ indexWord8ArrayAsWord256 ba off
+    | otherwise = error "deserialiseKey: not enough bytes for Word256"
 
-  deserialiseKey = error "deserialiseKey: Word256" -- TODO
+instance SerialiseValue Word256 where
+  serialiseValue w256 =
+    RB.RawBytes $ mkPrimVector 0 32 $ P.runByteArray $ do
+      ba <- P.newByteArray 32
+      P.writeByteArray ba 0 w256
+      return ba
+  deserialiseValue (RawBytes (PV.Vector off len ba))
+    | len >= 32  = indexWord8ArrayAsWord256 ba off
+    | otherwise = error "deserialiseValue: not enough bytes for Word256"
+  deserialiseValueN = deserialiseValue . mconcat -- TODO: optimise
+
+instance Arbitrary Word256 where
+  arbitrary = Word256 <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+  shrink w256 = [ w256'
+                | let i256 = toInteger w256
+                , i256' <- shrink i256
+                , toInteger (minBound :: Word256) <= i256'
+                , toInteger (maxBound :: Word256) >= i256'
+                , let w256' = fromIntegral i256'
+                ]
+
+{-# INLINE byteSwapWord256 #-}
+byteSwapWord256 :: Word256 -> Word256
+byteSwapWord256 (Word256 a3 a2 a1 a0) =
+    Word256 (byteSwap64 a0) (byteSwap64 a1) (byteSwap64 a2) (byteSwap64 a3)
+
+{-# INLINE indexWord8ArrayAsWord256 #-}
+indexWord8ArrayAsWord256 :: P.ByteArray -> Int -> Word256
+indexWord8ArrayAsWord256 (P.ByteArray ba#) (I# off#) =
+    Word256 (W64# (indexWord8ArrayAsWord64# ba# (off# +# 24#)))
+            (W64# (indexWord8ArrayAsWord64# ba# (off# +# 16#)))
+            (W64# (indexWord8ArrayAsWord64# ba# (off# +# 8#)))
+            (W64# (indexWord8ArrayAsWord64# ba# off#))
```

Comment:
Since you added it, you could also use the boxed `indexWord8ArrayAsWord64` wrapper here and below for `Word128`.